worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/MessageDistributor.java

```diff
@@ -17,51 +17,159 @@
+        this.targetQueueExclusive = targetQueueExclusive;
+        this.targetQueueAutoDelete = targetQueueAutoDelete;
+        this.targetQueueArgs = targetQueueArgs;
     }
-    
-    protected Set<DistributorWorkItem> getDistributorWorkItems() {
+
+    protected Set<DistributorWorkItem> getDistributorWorkItems(final Connection connection)
+    {
         final List<Queue> queues = queuesApi.getQueues();
 
         LOGGER.debug("Read the following list of queues from the RabbitMQ API: {}", queues);
 
         final Set<DistributorWorkItem> distributorWorkItems = new HashSet<>();
-        
-        for(final Queue targetQueue: queues.stream().filter(q -> !q.getName().contains(LOAD_BALANCED_INDICATOR))
-                .collect(Collectors.toSet())) {
+
+        final Set<Queue> targetQueues = getTargetQueues(connection, queues);
+
+        for (final Queue targetQueue : targetQueues) {
 
             final Set<Queue> stagingQueues = queues.stream()
                     .filter(q ->
                             q.getMessages() > 0 &&
                                     q.getName().startsWith(targetQueue.getName() + LOAD_BALANCED_INDICATOR))
                     .collect(Collectors.toSet());
-            
-            if(stagingQueues.isEmpty()) {
+
+            if (stagingQueues.isEmpty()) {
                 continue;
             }
-            
+
             LOGGER.debug("Creating a new DistributorWorkItem for target queue: {} and staging queues: {}", targetQueue, stagingQueues);
-            
+
             distributorWorkItems.add(new DistributorWorkItem(targetQueue, stagingQueues));
-            
+
         }
         return distributorWorkItems;
     }
+
+    private Set<Queue> getTargetQueues(final Connection connection, final List<Queue> queues)
+    {
+        // The set of target queues that already exist
+        final Set<Queue> targetQueuesAlreadyCreated = queues
+                .stream()
+                .filter(q -> !q.getName().contains(LOAD_BALANCED_INDICATOR))
+                .collect(Collectors.toSet());
+
+        // The set of target queues that need to be created (in the scenario where a worker is deployed initially with 0 instances its
+        // target queue will not be created, so when this happens we need to create the target queue, otherwise the messages will never
+        // get moved on from the staging queues)
+        final Set<String> targetQueuesToBeCreated = getTargetQueuesToBeCreated(queues, targetQueuesAlreadyCreated);
+
+        if (!targetQueuesToBeCreated.isEmpty()) {
+            LOGGER.info("Target queue(s) to be created: {}", targetQueuesToBeCreated);
+
+            // Create RabbitMQ channel
+            final Channel channel;
+            try {
+                channel = connection.createChannel();
+            } catch (final IOException e) {
+                throw new RuntimeException("IO Exception encountered trying to create a RabbitMQ channel", e);
```

Comment:
I think it would be better if you just added `throws IOException` to this `getTargetQueues()` function, and to the `getDistributorWorkItems()` function, and just let the exception propagate up.  It seems like a bit of an unnecessary hack to wrap it in a `RuntimeException`, given that it is within your control to have it propagate normally.